"ERROR:did not exit normally" -- Fails to install

Below is the error I get every time I try to build.  It happens for V1.13.9 and V1.13.6. 

```
2024-10-16 13:20:02,679:INFO:Library Manager: avr-debugger@1.2.0 has been installed! 
2024-10-16 13:20:02,687:INFO:Library Manager: Installing git+https://github.com/andre-stefanov/avr-interrupt-stepper @ 0.0.2 
2024-10-16 13:20:02,727:INFO:git version 2.41.0.windows.3 
2024-10-16 13:20:02,842:ERROR:Cloning into 'C:\Users\jack\AppData\Local\Temp\.pioOATFWGUI1.0.9-release+f0502f\.cache\tmp\pkg-installing-ye8sqzhl'... 
2024-10-16 13:20:04,941:ERROR:PackageException: Package version 0.0.4+sha.8deb3e2 doesn't satisfy requirements 0.0.2 based on PackageMetadata <type=library name=InterruptStepper version=0.0.4+sha.8deb3e2 spec={'owner': None, 'id': None, 'name': 'avr-interrupt-stepper', 'requirements': '0.0.2', 'uri': 'git+https://github.com/andre-stefanov/avr-interrupt-stepper'} 
2024-10-16 13:20:05,663:INFO:platformio build finished 
2024-10-16 13:20:05,664:ERROR:Did not exit normally
```

That log is showing a different error:
```
2024-10-17 12:52:07,549:ERROR:In file included from src\../Configuration.hpp:271:0,
2024-10-17 12:52:07,549:ERROR:                 from src\EPROMStore.cpp:6:
src\EPROMStore.cpp: In static member function 'static float EEPROMStore::getDECStepsPerDegree()':
src\../Configuration_adv.hpp:231:95: error: expected primary-expression before '/' token
     #define DEC_STEPS_PER_DEGREE (DEC_TRANSMISSION * DEC_STEPPER_SPR * DEC_SLEW_MICROSTEPPING / 360.0f)
                                                                                               ^
src\EPROMStore.cpp:431:29: note: in expansion of macro 'DEC_STEPS_PER_DEGREE'
     float decStepsPerDegree(DEC_STEPS_PER_DEGREE);  // Default value
                             ^~~~~~~~~~~~~~~~~~~~
src\EPROMStore.cpp:436:80: error: expected primary-expression before ';' token
         const float factor = SteppingStorageNormalized / DEC_SLEW_MICROSTEPPING;
                                                                                
src\EPROMStore.cpp: In static member function 'static void EEPROMStore::storeDECStepsPerDegree(float)':
src\EPROMStore.cpp:457:76: error: expected primary-expression before ';' token
     const float factor = SteppingStorageNormalized / DEC_SLEW_MICROSTEPPING;
```

Your `Configuration_local.hpp` looks broken, here:
```Cpp
...
#define DEC_STEPPER_SPEED              1200
#define DEC_SLEW_MICROSTEPPING         
#define DEC_GUIDE_MICROSTEPPING        
                                      ^ No values are here
// Is it going the wrong way?
#define DEC_INVERT_DIR  0
...
```

--- a/OATFWGUI/gui_logic.py
+++ b/OATFWGUI/gui_logic.py
@@ -95,6 +95,7 @@ class BusinessLogic:
         self.logic_state = LogicState()
 
         self.main_app = main_app
+        main_app.wCombo_fw_version.currentIndexChanged.connect(self.fw_version_combo_box_changed)
         main_app.wBtn_download_fw.setEnabled(True)
         main_app.wBtn_download_fw.clicked.connect(self.spawn_worker_thread(self.download_and_extract_fw))
         main_app.wBtn_select_local_config.clicked.connect(self.open_local_config_file)
@@ -209,6 +210,20 @@ class BusinessLogic:
         for pio_env_name in pio_environments:
             main_app.wCombo_pio_env.addItem(pio_env_name.nice_name)
         main_app.wCombo_pio_env.setPlaceholderText('Select Board')
+
+    @Slot()
+    def fw_version_combo_box_changed(self, idx: int):
+        if idx == self.logic_state.release_idx:
+            return  # Nothing changed, nothing to do
+        # Clear most state, if FW version is changed we want the user to go through the steps again
+        # (technically not necessary but can trip some users up)
+        log.debug('FW version changed, clearing some state')
+        self.logic_state.pio_envs.clear()
+        self.logic_state.pio_env = None
+        self.main_app.wSpn_download.setState(BusyIndicatorState.NONE)
+        self.main_app.wCombo_pio_env.clear()
+        self.main_app.wSpn_build.setState(BusyIndicatorState.NONE)
+        self.worker_finished()
 
     @Slot()
     def pio_env_combo_box_changed(self, idx: int):

--- a/OATFWGUI/gui_logic.py
+++ b/OATFWGUI/gui_logic.py
@@ -244,12 +244,27 @@ class BusinessLogic:
         # manually update GUI
         self.worker_finished()
 
-    def build_fw(self):
-        self.main_app.wSpn_build.setState(BusyIndicatorState.BUSY)
+    def do_hot_patches(self):
+        # Before logging anything, check that we need to do something
+        ini_lines = read_platformio_ini_file(self.logic_state)
+        bad_git_tag_re = re.compile(r'(github\.com.+)@')
+        if any(bad_git_tag_re.search(ini_line) for ini_line in ini_lines):
+            log.warning('Hot patching git tag specifiers!!!')
+            def patch_line(in_str: str) -> str:
+                if bad_git_tag_re.search(in_str):
+                    out_str = bad_git_tag_re.sub(r'\1#', in_str)
+                    log.warning(f'Replacing {in_str} with {out_str}')
+                    return out_str
+                else:
+                    return in_str
+            ini_lines = [
+                patch_line(line)
+                for line in ini_lines
+            ]
+            with open(Path(self.logic_state.fw_dir, 'platformio.ini').resolve(), 'w') as fp:
+                fp.writelines(ini_lines)
 
         if self.logic_state.env_is_avr_based():
-            # Before logging anything, check that we need to do something
-            ini_lines = read_platformio_ini_file(self.logic_state)
             # hard match the entire line
             # readline[s]() will always terminate a line with \n (and not \r\n on windows! :D)
             # https://docs.python.org/3.11/tutorial/inputoutput.html#methods-of-file-objects
@@ -264,6 +279,12 @@ class BusinessLogic:
                 ]
                 with open(Path(self.logic_state.fw_dir, 'platformio.ini').resolve(), 'w') as fp:
                     fp.writelines(ini_lines)
+
+    def build_fw(self):
+        self.main_app.wSpn_build.setState(BusyIndicatorState.BUSY)
+
+        # Hot patches, since we can't re-release an old firmware tag
+        self.do_hot_patches()
 
         config_dest_path = str(Path(self.logic_state.fw_dir, 'Configuration_local.hpp').resolve())
         if Path(config_dest_path) != Path(self.logic_state.config_file_path):

